httpevents/request.go

```diff
@@ -83,11 +85,11 @@ func (r *request) reset(req *http.Request, laddr string) {
 	r.query = req.URL.RawQuery
 	r.fragment = req.URL.Fragment
 	r.agent = req.UserAgent()
-	r.reqHeaders.set(req.Header)
+	r.reqHeaders.set(sanitizer.ReqHeaders(req.Header))
```

Comment:
If i'm following correctly, this will copy the headers even if we're doing a no-op pass through right?

But based on your benchmarking it just didn't really impact performance?

Reply:
Correct.  The alternative is down in log, we would update the headers there.  IIRC the benchmark I used (whatever one is on master now) probably didn't have any headers set anyways.

Reply:
Does something like this make any sense here:
```
	if sanitizer.reqHeaders != DefaultHeaderSanitizer {
		r.reqHeaders.set(sanitizer.ReqHeaders(req.Header))
	} else {
		r.reqHeaders.set(req.Header)
	}
```
this doesn't quite work since equality isn't defined...but i assume thats a solvable problem.

Reply:
You could figure something out, but if benchmarking doesn't point to this being a bottle neck I'd avoid that complexity.  I imagine most headers are ~2-6 key/values and thus should be pretty quick.

Reply:
Sounds good, makes sense, happy to avoid the complexity.